pandas/_libs/tslibs/timedeltas.pyi

```diff
@@ -50,9 +53,9 @@ class Timedelta(timedelta):
     @property
     def asm8(self) -> np.timedelta64: ...
     # TODO: round/floor/ceil could return NaT?
-    def round(self: _S, freq) -> _S: ...
-    def floor(self: _S, freq) -> _S: ...
-    def ceil(self: _S, freq) -> _S: ...
+    def round(self: _S, freq: str) -> _S: ...
+    def floor(self: _S, freq: str) -> _S: ...
+    def ceil(self: _S, freq: str) -> _S: ...
```

Comment:
could also be BaseOffset?

Reply:
I added most annotations based on the documentation (when available): it mentions only strings. `freq` ends up in `to_offset` so it should support `BaseOffset`.